### PR TITLE
PTL-587 Add how ranged dataserver works with examples

### PR DESCRIPTION
**Pull request by ShrutiTalewad**
Thank you for contributing to the documentation.

Your Jira ticket is:
PTL-587

Have you provided changes for all the relevant versions: next, 2022.4, 2023.1, etc?
Yes

Have you checked all new or changed links?
 Yes

Is there anything else you would like us to know?
No

For reference: 

- if you are an internal contributor:
  - We have an [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4)
  - We have a [style guide](https://www.notion.so/genesisglobal/Documentation-style-guide-5b04ec6fe12f4262b90d192effd8059b) 
- If you are an external contributor:
  - We have an [external contribution guide](../Type-of-contribution)

**This week's exciting excerpt from the style guide**
All our headings are in sentence case. For example:
- Really important information    **correct**
- Really Important Information    **not correct**

### CODE REVIEW

File: docs/03_server/02_data-server/03_advanced.md [hunk 1]
```diff
@@ -217,6 +217,97 @@ are always selected in the order they are specified in the index.
...
+    primaryKey(name = "GENESIS_PROCESS_MONITOR_BY_HOSTNAME", id = 1) {
+        PROCESS_HOSTNAME
+        PROCESS_NAME
+        MONITOR_NAME
+    }
+}
+```
+
+```text
+GENESIS_PROCESS_MONITOR.PROCESS_HOSTNAME,GENESIS_PROCESS_MONITOR.PROCESS_NAME,GENESIS_PROCESS_MONITOR.MONITOR_NAME
```
Comment (jacinpoz) [inline docs/03_server/02_data-server/03_advanced.md:239]: I think we can remove the table prefix so it is more concise.

File: docs/03_server/02_data-server/03_advanced.md [hunk 2]
```diff
@@ -217,6 +217,97 @@ are always selected in the order they are specified in the index.
 
 The `index` property can have unique and non-unique [indexes](../../../database/data-types/index-entities/#types)
 
+Examples when: `numKeyFields > 1`
+
+Range is based on sorted records when numKeyFields is greater than 1
```
Comment (jacinpoz) [inline docs/03_server/02_data-server/03_advanced.md:222]: maybe rephrase this to something like:
"The range configuration returns a set of sorted records based on the index definition and the number of fields involved (i.e. numKeyFields configuration)"

File: docs/03_server/02_data-server/03_advanced.md [hunk 3]
```diff
@@ -217,6 +217,97 @@ are always selected in the order they are specified in the index.
...
+For below table and data which is sorted already, we have shown a couple of examples of ranged query with its result
```
Comment (jacinpoz) [inline docs/03_server/02_data-server/03_advanced.md:223]: I would rephrase it like:

"Let's see a couple of examples of ranged queries with their respective results in CSV format. The table definition with associated data sorted by its primary key fields is shown below.

File: docs/03_server/02_data-server/03_advanced.md [hunk 4]
```diff
@@ -217,6 +217,97 @@ are always selected in the order they are specified in the index.
...
+                    processName = "process_a",
+                    monitorName = "monitor_a"
+                )
+            }
+        }
+    }
+```
+Result:
+```text
+localhost,process_a,monitor_a
```
Comment (jacinpoz) [inline docs/03_server/02_data-server/03_advanced.md:304]: We could add the CSV headers here as well so it is more clear.

**Review by kievitsp — APPROVED** [review]

**Review by wjhendry — APPROVED** [review]
